test/instruction/instructions/adcs.test.ts

```diff
@@ -0,0 +1,51 @@
+import { Word } from 'types/binary'
+import { Register, Registers } from 'board/registers'
+import { Memory } from 'board/memory'
+import { AdcsInstruction } from 'instruction/instructions/adcs'
+import { AddsRegistersInstruction } from 'instruction/instructions/adds'
+
+const registers = new Registers()
+const memory = new Memory()
+
+const adcsInstruction = new AdcsInstruction()
+const addsInstruction = new AddsRegistersInstruction()
+
+beforeEach(function () {
+  registers.reset()
+  registers.writeRegister(Register.R1, Word.fromUnsignedInteger(0xffffffff))
+  registers.writeRegister(Register.R2, Word.fromUnsignedInteger(0x01))
+  registers.writeRegister(Register.R3, Word.fromUnsignedInteger(0x01))
+  registers.writeRegister(Register.R4, Word.fromUnsignedInteger(0x00))
+})
+
+describe('test encodeInstruction function for ADCS', () => {
+  it('should create correct opcode for ADCS R1, R3', () => {
+    let opcode = adcsInstruction.encodeInstruction(['R1', 'R3'], {})
+    expect(opcode.toBinaryString()).toEqual('0100000101011001')
+  })
+
+  it('should create correct opcode for ADCS R1, R1, R3', () => {
+    let opcode = adcsInstruction.encodeInstruction(['R1', 'R1', 'R3'], {})
+    expect(opcode.toBinaryString()).toEqual('0100000101011001')
+  })
+
+  it('should throw an error for different params 0 and 1', () => {
+    let registerArray = ['R1', 'R2', 'R3']
+    expect(() => adcsInstruction.encodeInstruction(registerArray, {})).toThrow()
+  })
+
```

Comment:
no test with carry set provided

Reply:
My bad comment is on wrong section. you are right for encoding ther is no difference.
However, for the execute test a test without carry would not hurt.

Reply:
Additionally you could use registers.setFlag(Flag.C, true) instead of the addition which results in carry

Reply:
Ok you have convinced me. I'll add/change that, respectively. Thx for your attentive eye!